text/0000-volta-run.md

```diff
@@ -0,0 +1,72 @@
+- Feature Name: volta_run
+- Start Date: 2019-09-14
+- RFC PR: (leave this empty)
+- Volta Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add a `volta run` command, allowing users to execute tools with an override on the detected platform.
+
+# Motivation
+[motivation]: #motivation
+
+Volta currently will automatically detect the platform (Node runtime and yarn version) based on the context of where you are in the file system. However, there are occasionally times when you want to run a command using a different version of Node or Yarn than what Volta detects, without changing the pinned or default versions. To support those use-cases, we should provide a command that allows users to execute a command with a platform specified on the command-line.
+
+# Pedagogy
+[pedagogy]: #pedagogy
+
+We will need to document the existence of `volta run` and how the users can specify the versions of Node and Yarn when running a command. This will be provided on the command line by StructOpt, however it should also be explained in detail in the documentation. We should also document how the overall platform is resolved if the user specifies only one of Node or Yarn, so that the behavior is clear to users.
+
+# Details
+[details]: #details
+
+The new command will have the structure:
+
+```
+volta run [--node <version>] [--yarn <version>] <command>
```

Comment:
Since we ultimately _plan_ to allow custom `npm` version pinning, we should at least consider (here, in unanswered questions, a future changes section) what pinning the `npm` version would look like also.

In addition, there are quite a few _other_ tools / registries that we _might_ eventually want to support (entropic, pnpm, etc). Do we need an option for each?

Reply:
Can you add an example command here (just below the usage)? It is not obvious how the command is specified and we don't give an example until a few sections later.

Reply:
Done and done. I added an example immediately after the usage, and a new section "Additional Package Managers" to cover how it would look as we add support for other parts of the platform.